### DISCUSSION TITLE
Enums for states/conditions/etc

### DISCUSSION THREAD

**Discussion by AnssiAhola** [joalla/discogs_client#28] [Ideas] [answered]
Could be useful for preventing mistyping or is this best left for the clients user to implement?

Example
```python
class Condition(Enum):
    MINT = 'Mint (M)'
    NEAR_MINT = 'Near Mint (NM or M-)'
    ...
```
```python
class State(Enum):
    DRAFT = 'Draft'
    FOR_SALE = 'For Sale'
```
```python
listing.state = State.DRAFT
listing.condition = Condition.NEAR_MINT
```

**Comment by JOJ0** [comment, marked as answer]
Does the discogs.com API itself accept anything to be put in these fields? Can I just write 'Mintest!!! (M)' and it will show as that in my inventory or won't a request like that be accepted by the API? I haven't ever tried to work with sales listings and stuff via the API yet (But soon will start using it trough discogs_client thanks to your PR's! :-))

**Reply by AnssiAhola** [reply]
It seems to return `422: invalid condition` if for example the condition is not valid. [Valid condition strings](https://www.discogs.com/developers#page:marketplace,header:marketplace-new-listing)
And `422: invalid sleeve_condition` if sleeve_condition is not valid string, etc

**Reply by alifhughes** [reply]
@AnssiAhola Seems like a good thing to do to me, would welcome the contribution!

**Reply by alifhughes** [reply]
This has been included in the release version 2.3.10 - link to download it via PyPI [here](https://pypi.org/project/python3-discogs-client/2.3.10/)
